app/helpers/abstract_xml_helper.rb

```diff
@@ -38,23 +43,30 @@ def xml_to_html(xml_text, preserve_lb=true, flatten_links=false)
     end
     # get rid of line breaks within other html mark-up
     doc.elements.delete_all("//table//lb")
+
     # convert line breaks to br or nothing, depending
     doc.elements.each("//lb") do |e|
       lb = REXML::Element.new('span')
       lb.add_text("")
       lb.add_attribute('class', 'line-break')
 
       if preserve_lb
-        if e.attributes['break'] == "no"
-          lb.add_text('-')
-        end
-        lb.add_element(REXML::Element.new('br'))
+        e.replace_with(REXML::Element.new('br'))
       else
-        unless e.attributes['break']=="no"
-          lb.add_text(' ')
+        if params[:action] == "read_work"
+          if e.attributes['break'] == "no"
+            lb.add_text('')
+          end
+        else
+          if e.attributes['break'] == "no"
+            lb.add_text('-')
+          end
         end
+
+        lb.add_element(REXML::Element.new('br'))
```

Comment:
If `preserve_lb==false`, we need to stop adding `br` tags, so this line needs to be deleted.